### ISSUE TITLE
Arbitrary command execution

### ISSUE THREAD

**Issue by zdykstra** [davidscholberg/goblocks#14]
I have a quick-and-dirty module that executes an arbitrary command, grabs the first line by splitting on '\n' and then outputs it. I use it for grabbing the currently playing song information from Deadbeef. If something like this can be considered for inclusion, I'll clean the code up and add in a few extra features if you can think of anything you'd like to see.

**Comment by davidscholberg** [comment]
Yeah, I've been considering something like this.

It turns out that @mroote has already implemented this functionality in the [command branch of his fork](https://github.com/mroote/goblocks/commits/command). I'm not even sure if he'll get a notification for this mention, but nonetheless I'll give him a day or two to comment on this ticket if he has any input. @mroote, specifically, I was wondering if you were at all interested in getting your changes merged upstream. I see you have a couple other nice fixes for goblocks, and I welcome any contributions you care to send!

@zdykstra you can, of course, submit a pull request whenever you like. Or maybe you'll decide that you like @mroote's implementation better, in which case we can pull his changes. Either way, I'd say let's give @mroote some time to respond.

**Comment by mroote** [comment]
Hi @davidscholberg and @zdykstra,

I've implemented this feature in my branch of goblocks and would be happy to open a PR.  I wasn't sure if you were looking to add this feature as I know in the README you are trying to keep all the different blocks coded in Go to keep the application performant.

**Comment by mroote** [comment]
I've opened a PR with my command branch #15

**Comment by davidscholberg** [comment]
@mroote excellent, thank you!

Yeah, I would ideally like everything to be in pure go, but i think it's perfectly reasonable to have an arbitrary command module that people can use if they so choose.

Generally, if you have any thoughts or suggestions for features, you can feel free to open an issue or PR any time, and I'm sure we can work something out.

@zdykstra, feel free to take look at PR #15 to see if it meets your needs, and let us know if you have any feedback.

**Comment by zdykstra** [comment]
The only issue I see (that's also present in the module I wrote) is that executing a command that takes arguments is impossible. I ended up creating a little wrapper shell script to run the following:

/opt/deadbeef/bin/deadbeef --nowplaying-tf "%artist% - %title%"

I'm not sure if it's worth the effort to break up the command line into the exec target and the arguments that should passed to it.

**Comment by mroote** [comment]
This shouldn't be too difficult to implement, I think it would be useful to support adding flags.  Hopefully I'll have to some time this evening to test this out.

**Comment by mroote** [comment]
I've added parsing of any arguments to my PR.  I played around with your example command but parsing with the quotation marks in a command was proving difficult.   You are probably better off executing complex commands from a simple shell script, that way supporting things like pipe characters would work properly.

I also added an example to the goblocks-full.yml config file and just a bit of formatting.
